cwl/src/main/scala/cwl/CwlCodecs.scala

```diff
@@ -1,22 +1,34 @@
 package cwl
 
+import cats.data.NonEmptyList
 import io.circe._
 import io.circe.parser._
 import io.circe.shapes._
 import io.circe.generic.auto._
 import eu.timepit.refined.string._
 import io.circe.refined._
 import io.circe.literal._
+import shapeless.Coproduct
+import cats.syntax.either._
+import cats.syntax.show._
+
 
 object CwlCodecs {
   import Implicits._
 
-  //According to automatic derivation, these instances should not be required.  But
-  //removing these breaks decodeCwl, so...
-  implicit val wfD = implicitly[Decoder[Workflow]]
-  implicit val cltD = implicitly[Decoder[CommandLineTool]]
-
-  def decodeCwl(in: String): Either[Error, Cwl] = decode[Cwl](in)
+  def decodeCwl(in: String): Either[NonEmptyList[String], Cwl] = {
+    //try to parse both and combine errors if they fail
+    (decode[Workflow](in), decode[CommandLineTool](in)) match {
```

Comment:
can you `lazy` the `CommandLineTool` version to save a parse in the likely event you get a `Workflow`?

Reply:
If there is a way I'm not aware of how to do that.  I could do one-then-the-other but I lose pattern matching niceness.

Plus I don't think it's terribly wasteful to just try both. :man_shrugging: